src/dmd/optimize.d

```diff
@@ -1167,15 +1178,15 @@ extern (C++) Expression Expression_optimize(Expression e, int result, bool keepL
         }
     }
 
-    scope OptimizeVisitor v = new OptimizeVisitor(result, keepLvalue);
-    Expression ex = null;
-    v.ret = e;
+    scope OptimizeVisitor v = new OptimizeVisitor(e, result, keepLvalue);
 
     // Optimize the expression until it can no longer be simplified.
-    while (ex != v.ret)
+    while (1)
```

Comment:
`while(true)` please.

Reply:
`while (1)` is extensively used in dmd. `while (true)` is used 5 times.

Reply:
Then maybe it's time to change the others in another PR.  Let's try to get rid of some of this legacy cruft.

Reply:
No, I prefer `while (1)`. Short & sweet.

Reply:
The reason I redid that loop in the first place is it took me some time to verify that the loop was correct. I redid it so that it was quicker to see that it was correct. The latter is the kind of refactorings we need in dmd - ones that make the code flow in a simpler, less convoluted manner.

The same goes for redoing `expandVar()`. The use of nested functions makes the logic flow much more understandably.

Reply:
For another example, see line 225. I relaid out the fields and the constructor parameters and the `Params:` to match the sequence of the parameters in the declaration. These little things just make the code easier to read.